php/libraries/FilesUploadHandler.class.inc

```diff
@@ -0,0 +1,129 @@
+<?php
+/**
+ * An interface containing methods for classes used to handle file
+ * uploads from a client to a server.
+ *
+ * PHP Version 7
+ *
+ * @category Main
+ * @package  Main
+ * @author   Nicolas Brossard <nicolasbrossard.mni@gmail.com>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+
+/**
+ * Handles file uploads from a client to a server. The methods listed in this
+ * interface were created in order to encapsulate the usual things that happen
+ * when a set of files is uploaded to a server, namely:
+ *
+ * 1. Validation of all uploaded files
+ * 2. Transfer of all files to a designated (permanent) location on the
+ *    server.
+ * 3. Actual processing of the uploaded files (e.g. update of database
+ *    tables, etc...)
+ *
+ * @category Main
+ * @package  Main
+ * @author   Nicolas Brossard <nicolasbrossard.mni@gmail.com>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+interface FilesUploadHandler
+{
+    /**
+     * Returns the properties associated to each file uploaded on the
+     * server in an array identical to $_FILES.
+     *
+     * @return array the properties for each file uploaded (one array for
+     *         each uploaded file, see $_FILES).
+     */
+    public function getFiles() : array;
+
+    /**
+     * Validates the set of uploaded files. If a given file is invalid, the
+     * validation message associated to it will be stored in the array returned
+     * by getErrors().
+     *
+     * @return bool true if the set of files is valid and should be processed, false
+     *         otherwise.
+     */
+    public function isValid() : bool;
+
+    /**
+     * Returns any validation or processing errors associated to each uploaded file.
+     *
+     * @return array list of error messages for each uploaded file.
+     */
+    public function getErrors() : array;
+
+    /**
+     * Returns the final locations of all files that were successfully moved. This
+     * method will not return anything meaningful until the move operation is done.
+     *
+     * @return SplFileInfo[] final locations of all files that were successfully
+     *                       moved.
+     */
+    public function getMovedFiles() : array;
+
+    /**
+     * Main entry point for classes implementing this interface. Processes all
+     * uploaded files.
+     *
+     * @return NULL.
+     */
+    public function processAll();
+
+    /**
+     * Takes a file that was uploaded on the server and moves it from its temporary
+     * location to a permament location.
+     *
+     * @param string $tmpFileName temporary path to the uploaded file on the server.
+     *
+     * @return NULL.
+     */
+    public function move(string $tmpFileName);
```

Comment:
UploadedFileInterface would be better than a string here too, I think, since then you have more metadata than the file name to make your decisions on (and the moveTo() function that's already part of the PSR7 interface would be helpful for implementations.)